### ISSUE TITLE
Same Origin Response Error - HTTP 302 header doesn't appear

### ISSUE THREAD

**Issue by Comancheroo** [beefproject/beef#1494]
Hello,

I'm using Rider.
I get an error when there is redirection in the page result.
HTTP/1.1 302 Moved Temporarily
Location:othersite.com
___________________

**Example 2 Request;**

GET /index.php HTTP/1.1
Host: x.com:80

**HTTP/1.1 200 OK**
Rescode:200
Restext: success

**HTTP Header: OK**
_______________
GET /menu.php HTTP/1.1
Host: x.com:80

**HTTP/1.1 302 Moved Temporarily**
Rescode:0
Restext: error

**HTTP Header: NULL**
_______________

I want to see Response(only header).
Because HTTP headers > **Location**: url very important for me. Is it possible?
Thanks.

**Comment by bcoles** [comment]
Hi @Comancheroo 

tl/dr: unfortunately, no.

For same-origin requests, the browser automatically follows the redirect. I'm not sure if it's possible to get the `Location` header from the response. BeEF doesn't support this as it follows the redirect.

For cross-origin requests with CORS it's a little more complicated. The short answer is the destination server must use the `Access-Control-Expose-Headers` header to expose the `Location` header to the CORS client.

From [HTML5Rocks](https://www.html5rocks.com/en/tutorials/cors/) :

> Access-Control-Expose-Headers (optional) - The XMLHttpRequest 2 object has a getResponseHeader() method that returns the value of a particular response header. During a CORS request, the getResponseHeader() method can only access simple response headers. Simple response headers are defined as follows:
>
> *    Cache-Control
> *    Content-Language
> *    Content-Type
> *    Expires
> *    Last-Modified
> *    Pragma
>
> If you want clients to be able to access other headers, you have to use the Access-Control-Expose-Headers header. The value of this header is a comma-delimited list of response headers you want to expose to the client.

For cross-origin requests with Flash (using *crossdomain.xml*) - I'm not sure if this is possible. I did some quick tests with [CrossSiteContentHijacking](https://github.com/nccgroup/CrossSiteContentHijacking) and wasn't able to view the headers as Flash followed the redirect.

**Comment by DeezyE** [comment]
Closed. Solved
